### ISSUE TITLE
Device control works in simulator but not other Assistant surfaces

### ISSUE THREAD

**Issue by cketcham** [actions-on-google/actionssdk-smart-home-nodejs#22]
I'm having a problem that after the sync request and my devices are added to the Google home app, I can't control them. My Google home just responds with a generic message that it can't understand and it never requests anything else from the server (besides the initial sync).

For a while when I first set up a server and connected it to Google home, I was able to send commands.. but it stopped working all of a sudden and I can't figure out if it's something I did, or something else.

Do you have any suggestions on how to debug it?

**Comment by Fleker** [comment]
Your test agent expires after three days. Make sure you redo the test with `gactions test`.

**Comment by cketcham** [comment]
I did redo it, but it still didn't work

**Comment by Fleker** [comment]
Can you try re-linking your devices? Are you seeing any commands (EXECUTE, QUERY) going to your server?

**Comment by dooyawoo** [comment]
@cketcham how to sync requeset.I cannot find my app  in the list of providers.This is the list of providers in my google home.
![img_0257](https://user-images.githubusercontent.com/6778877/28949469-951a88f8-78ef-11e7-95c9-7fcc0502dd3a.JPG)

**Comment by cketcham** [comment]
Yeah, I start the test agent, open the Google home app,  choose my service. At that point I see the auth steps happening on my server and then I get the sync command. After that, it lists my devices and I can assign them to rooms. But if I try to interact with them on my Google home, nothing happens. I don't get any other commands on my server.

**Comment by dooyawoo** [comment]
i have the same issue!
I can control device using the web smart home provider 
the logs like this:
POST /smart-home-api/exec 200 1.354 ms - 636
smartHomeExec executedDevice { '1': 
   { states: { on: true, online: true, brightness: 80, color: [Object] },
     properties: 
      { type: 'action.devices.types.LIGHT',
        traits: [Array],
        name: [Object],
        willReportState: false,
        roomHint: 'living room',
        deviceInfo: [Object],
        customData: [Object],
        id: '1' } } }
POST /smart-home-api/exec 200 1.484 ms - 635
the pic:
<img width="424" alt="2" src="https://user-images.githubusercontent.com/6778877/28992684-c91c4c7c-79d5-11e7-8c3d-77a65ced548c.png">
but the test Simulator show:
<img width="393" alt="3" src="https://user-images.githubusercontent.com/6778877/28992691-e9d8038e-79d5-11e7-9697-48dda2679f1e.png">

**Comment by Fleker** [comment]
Did you name the light 'red lamp' before you synced the devices when you linked it to the Google Home app?

**Comment by dooyawoo** [comment]
@Fleker Now. It is ok without any change!

**Comment by cketcham** [comment]
@Fleker I think I have a different problem. I didn't make any changes, but now the simulator is working. I can issue the command `turn on the projector` to the simulator and it responds correctly and I see `EXEC` on my server. But if I issue commands to my google home (with the same account) it says "Sorry I don't know how to help with that yet"

**Comment by Fleker** [comment]
Can you check again and see if the problem continues? Make sure you rerun the test so that the agent hasn't expired.

**Comment by cketcham** [comment]
@Fleker I tested again. I noticed if I just run `gactions test` and open the simulator I get `I'm sorry, there was an error and I'm unable to control your home device.` when trying to interact with a device and I don't see any communication to my server. If I ask my google home to interact with a device it says `Sorry, I don't know how to help with that yet.`. If I go to the google home app and unlink my account and then relink my account I can then control devices using the simulator, but my google home still gives me the same error.

**Comment by Fleker** [comment]
Yes you will need to relink your account as the test agent has expired and
you will need to resync devices.
Are you sure you're pronouncing the voice commands correctly to your Home?
Is it linked to your same Google account?

**Comment by cketcham** [comment]
Ok, relinking makes sense. I checked the action log for my account (myactivity.google.com) and it's appearing there with the correct command. I have two accounts linked to my Google home, could that make a difference?

This is the error log that I see in my activity for the command I tried
```
Said turn on the projector
Sorry, I don't know how to help with that yet.
com.google.homeautomation
I'm sorry, there was an error and I'm unable to control your home device.
```

My Google home said "Sorry, I don't know how to help with that yet." But looking at that error it does look like it's getting the command, but having an error? I don't see anything on my server though. And I'm trying the command after I relink my account.

**Comment by Fleker** [comment]
The multiple accounts shouldn't make a difference. What is the name of your device? Can you send a more precise phrase. "Turn on `deviceName`" and see if that executes correctly?

You don't see any command on your server?

**Comment by cketcham** [comment]
I have two devices specified. This is the json for the devices for the user I'm testing with

```
{
  "-KqdKpgq8CS85jiGl15x" : {
    "properties" : {
      "customData" : {
        "smartHomeProviderId" : "FkldJVJCmDNSaoLkoq0txiz8Byf2Hr"
      },
      "deviceInfo" : {
        "hwVersion" : "1.0",
        "manufacturer" : "Smart Home Provider",
        "model" : "g1337",
        "swVersion" : "1.0.11"
      },
      "name" : {
        "defaultNames" : [ "Smart Light" ],
        "name" : "Smart Light"
      },
      "traits" : [ "action.devices.traits.OnOff", "action.devices.traits.Brightness" ],
      "type" : "action.devices.types.LIGHT",
      "willReportState" : false
    },
    "states" : {
      "brightness" : 20,
      "color" : {
        "name" : "cerulian",
        "spectrumRGB" : 31655
      },
      "on" : false,
      "online" : true
    }
  },
  "-KqdMUdXOccuGJ7QET4t" : {
    "properties" : {
      "customData" : {
        "smartHomeProviderId" : "FkldJVJCmDNSaoLkoq0txiz8Byf2Hr"
      },
      "deviceInfo" : {
        "hwVersion" : "1.0",
        "manufacturer" : "Smart Home Provider",
        "model" : "g1337",
        "swVersion" : "1.0.11"
      },
      "name" : {
        "defaultNames" : [ "Projector" ],
        "name" : "Projector",
        "nicknames" : [ "Projector" ]
      },
      "roomHint" : "living room",
      "traits" : [ "action.devices.traits.OnOff" ],
      "type" : "action.devices.types.SWITCH",
      "willReportState" : false
    },
    "states" : {
      "on" : true,
      "online" : true
    }
  }
}
```

I tried both "turn on projector" and "turn on smart light" with my google home and it responded with "Sorry, I'm not sure how to help with that." And I see no communication with my server. But if I try the same commands on the simulator it responds correctly and I see the state change on my server.

**Comment by pantongren** [comment]
@cketcham Sir, Have you solved your last issue?  I can get the correct response only by simulator too. Have you try it on Pixel's Google Assistant  or other phone's  Allo App? I don't have a Pixel, so I can only test it in Allo App which contains Google Assitant features. I tried "turn on light" with my Allo App and it responded with "Sorry,I can't quite control that yet". thanks

**Comment by Fleker** [comment]
@pantongren what version of Android is your phone? Assistant should be on all Android 6.0+ devices.

@cketcham What accounts on your Home? Can you determine which account is seeing these Smart Home commands (in [My Activity](https://myactivity.google.com/myactivity))? Is it the same account as your project?

Renaming issue title to be more specific

**Comment by cketcham** [comment]
@Fleker yes, it's the same account

**Comment by Fleker** [comment]
So can you visit your My Activity and get the Google Home transcriptions?

**Comment by cketcham** [comment]
@Fleker yes. I tested again with the simulator and my google pixel and both of those work and I see the requests hit my server, and I see the transcriptions in my activity "Said turn on the smart light" with the response "OK, turning the Smart Light on." But if I ask my google home I see the response "Sorry, I don't know how to help with that. But I'm still learning."

**Comment by Fleker** [comment]
And what does it say in [your activity](https://myactivity.google.com/myactivity) for that command?

**Comment by Fleker** [comment]
From @pantongren in #26 :

I tried to reset my Google Home Device a few days ago,but failed to set it up during the last step of configuration. So, I can only use the web simulator or Allo App. I have tried "turn on light" with Allo App, it replied "I can't quite control that yet". So, it works only by the web simulator. I'm just testing this given sample which contains the whole OAuth process. I have done no modifacation with the sample OAuth,thanks.

**Comment by pantongren** [comment]
@Fleker  I'm sorry I did not see your reply of asking the version of my android phone. My phone is android 6.0. I can interact with the Assistant in Allo App,but it failed to handle my  "turn on light" command with the response "I can't quite control that yet". I am now in Mainland China.

**Comment by Fleker** [comment]
Can you try invoking the Assistant directly? Is the issue related to your device language perhaps?
From [the help page](https://support.google.com/allo/answer/7276628?hl=en) "The Google Assistant in Allo is available in English, French, German, Hindi, Indonesian, Japanese, Portuguese, and Spanish."

I don't know whether being in Mainland China would affect the ability of the Assistant to complete queries.

**Comment by pantongren** [comment]
@Fleker  You mean can I invoke the Assistant by saying " Ok Google" in the home menu? I think this feature is not available in Allo,  Allo is a 3rd part App instead of a system App. Maybe some phones like Google Pixel that built-in Google Assitant can invoke the Assistant directly. I have set the system language to English(US), but it still doesn't work. Is the issue related to my registration phone number?  using  phone number to sign up is required in Allo. I haven't use the same phone number signed up for Google Account to sign up in Allo.

**Comment by Fleker** [comment]
@pantongren do you have Google Assistant on your device, or only Allo?

I'm not sure that Allo supports home control.

**Comment by cketcham** [comment]
@Fleker In https://github.com/actions-on-google/actionssdk-smart-home-nodejs/issues/22#issuecomment-324119174 I put what it says for my activity. Was there something else that it says that you wanted to know?

**Comment by cketcham** [comment]
Here's a screenshot of what I see in my activity when I ask my google home to interact with one of my devices.

<img width="639" alt="screen shot 2017-08-31 at 6 11 58 pm" src="https://user-images.githubusercontent.com/16722/29947500-f693ddd0-8e77-11e7-8a79-d7a3b6d813be.png">

Here is a screenshot when I used the simulator to interact with a device (which is working correctly).
<img width="634" alt="screen shot 2017-08-31 at 6 14 24 pm" src="https://user-images.githubusercontent.com/16722/29947571-485c602e-8e78-11e7-8de1-3f07e8f5a146.png">

Sorry it's for different devices, these are just two that I found where they weren't grouped together with other requests.

**Comment by Fleker** [comment]
Thanks. I'll forward these issues along @cketcham. Have you tested on an Android phone's Assistant?

**Comment by cketcham** [comment]
yes, it works correctly in the simulator and google assistant on my phone, but not my google home.

**Comment by pantongren** [comment]
@Fleker I have only a Allo. Sorry Sir,  the simulator didn't work just now.  When I saved my draft and then clicked on"test draft", it jumped to the simulator page but responded with"failed to enable the simulator", and then I tried "START TESTING" in the web simulator page, it responded with"start test failed". What happend to that?  If I choose an old Version of my smart home App to test, it works fine. Why suddenly can't test my draft App? thanks.
